src/main/java/fi/istrange/traveler/resources/AuthResource.java

```diff
@@ -36,21 +37,8 @@ public DefaultJwtCookiePrincipal login(
             @NotNull UserCredentialsView userCredentialsView,
             @Context DSLContext database
     ) {
-        if (userCredentialsView.getName() == null || userCredentialsView.getPassword() == null) {
-            throw new WebApplicationException(422);
-        }
-
-        Optional<Record> result = database.select()
-                .from(Tables.USER_CREDENTIALS)
-                .where(Tables.USER_CREDENTIALS.USERNAME.equal(userCredentialsView.getName()))
-                .fetchOptional();
-
-        if (
-                !result.isPresent() ||
-                !userCredentialsView.getPassword().equals(result.get().getValue(Tables.USER_CREDENTIALS.PASSWORD)) ||
-                !result.get().getValue(Tables.USER_CREDENTIALS.ACTIVE)
-        ) {
-            throw new NotAuthorizedException("Invalid credentials");
+        if (!auth.authenticate(userCredentialsView.getName(), userCredentialsView.getPassword(), database)) {
+            return null;
```

Comment:
I missed the fact, that the authenticate can throw the 422 exception.

Maybe it's better to add the throws signature to authenticate. And then propagate the exception to the resource and handle/throw it there. As I see this conditional statement will never be executed and returning the null value, that will never be returned, is confusing a bit.